src/plugins/data/public/search/expressions/build_tabular_inspector_data.ts

```diff
@@ -38,14 +39,30 @@ import { createFilter } from './create_filter';
  */
 export async function buildTabularInspectorData(
   table: TabbedTable,
-  queryFilter: { addFilters: (filter: any) => void }
+  {
+    queryFilter,
+    deserializeFieldFormat,
+  }: {
+    queryFilter: { addFilters: (filter: any) => void };
+    deserializeFieldFormat: FormatFactory;
+  }
 ) {
   const aggConfigs = table.columns.map((column) => column.aggConfig);
   const rows = table.rows.map((row) => {
     return table.columns.reduce<Record<string, FormattedData>>((prev, cur, colIndex) => {
       const value = row[cur.id];
-      const fieldFormatter = cur.aggConfig.fieldFormatter('text');
-      prev[`col-${colIndex}-${cur.aggConfig.id}`] = new FormattedData(value, fieldFormatter(value));
+
+      let format = cur.aggConfig.toSerializedFieldFormat();
+      if (Object.keys(format).length < 1) {
+        // If no format exists, fall back to string as a default
+        format = { id: 'string' };
+      }
```

Comment:
This fallback behavior existed previously in `agg.fieldFormatter()` so we need to replicate it here or empty serialized formats (`{}`) don't receive any formatting, like `-` for an empty value.